src/generate-types.ts

```diff
@@ -423,6 +443,10 @@ export function run(options: Options) {
           required: ['status', 'value'],
           additionalProperties: false
         };
+        if (!oautil.isReferenceObject(response) && response.headers) {
+          schema.properties!.headers = generateHeadersSchemaType(response.headers);
+          schema.required!.push('headers');
```

Comment:
so to check we would only need to provide a headers object in the response if there are response headers defined in the spec?

Reply:
I guess I could also check if there's actually any headers declared in there, so we don't end up having to set empty objects as `headers` 👍